pkg/ocm/clustertransfer/cluster_transfer.go

```diff
@@ -0,0 +1,276 @@
+// in the response then check if a secret update is required, and if so, perform the update.
+func (c *Controller) requestDataAndUpdateSecret(endpoint string) {
+	data, err := c.requestClusterTransferWithExponentialBackoff(endpoint)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to pull cluster transfer: %v", err)
+		httpErr, ok := err.(insightsclient.HttpError)
+		// HTTP error received
+		if ok {
+			reason := strings.ReplaceAll(http.StatusText(httpErr.StatusCode), " ", "")
+			c.updateStatus(false, msg, reason, &httpErr)
+			return
+		}
+		// we are probably in disconnected environment
+		klog.Errorf(msg)
+		c.updateStatus(true, "", "", nil)
+		return
+	}
+
+	// there's no cluster transfer for the cluster - HTTP 204
+	if len(data) == 0 {
+		klog.Info("No cluster transfer exists")
+		c.updateStatus(true, "no cluster transfer content", "NoData", nil)
+		return
+	}
+
+	// TODO somewhere here we need to check if the ClusterTransfer is Accepted
+
+	var statusMsg string
+	updating, err := c.isUpdateRequired(data)
+	if err != nil {
+		statusMsg = fmt.Sprintf("pull-secret check failed: %v", err)
+		klog.Errorf(statusMsg)
+		c.updateStatus(false, statusMsg, "DataCorrupted", nil)
+		return
+	}
+	if updating {
+		err = c.updatePullSecret(data)
+		if err != nil {
+			statusMsg = fmt.Sprintf("Failed to update pull-secret: %v", err)
+			klog.Errorf(statusMsg)
+			c.updateStatus(false, statusMsg, "UpdateFailed", nil)
+			return
+		}
+		statusMsg = "pull-secret successfully updated"
+		klog.Info(statusMsg)
+	} else {
+		statusMsg = "no new data received"
+		klog.Info(statusMsg)
+	}
+	c.updateStatus(true, statusMsg, "", nil)
+}
+
+// isUpdateRequired checks if an update of the pull-secret is required or not.
+func (c *Controller) isUpdateRequired(newData []byte) (bool, error) {
+	pullSecret, err := c.getPullSecret()
+	if err != nil {
+		return false, err
+	}
+	c.pullSecret = pullSecret
+
+	newPullSecret, err := unmarshallToPullSecretContent(newData)
+	if err != nil {
+		return false, err
+	}
+	originalPullSecret, err := unmarshallToPullSecretContent(pullSecret.Data[v1.DockerConfigJsonKey])
+	if err != nil {
+		return false, err
+	}
+	return !isUpdatedPullSecretContentSame(*originalPullSecret, *newPullSecret), nil
+}
+
+// updatePullSecret creates a JSON merge patch of existing and new pull-secret data.
+// The result of the patch is used for pull-secret data update.
+func (c *Controller) updatePullSecret(newData []byte) error {
+	if c.pullSecret == nil {
+		ps, err := c.getPullSecret()
+		if err != nil {
+			return err
+		}
+		c.pullSecret = ps
+	}
+	existingData := c.pullSecret.Data[v1.DockerConfigJsonKey]
+
+	updatedData, err := jsonpatch.MergePatch(existingData, newData)
+	if err != nil {
+		return err
+	}
+
+	c.pullSecret.Data[v1.DockerConfigJsonKey] = updatedData
+	_, err = c.coreClient.Secrets("openshift-config").Update(c.ctx, c.pullSecret, metav1.UpdateOptions{})
+	if err != nil {
```

Comment:
Can't you just return err?

Reply:
What do you mean please? I am returning the error if it is not `nil`

Reply:
it is just an idea, 'cause I guess that if there is no error, `err` will be `nil`... right?

Reply:
Ahh OK now I understand. Yes it should be possible. I think this is more explicit but I don't have a strong opinion